demonstrations/tutorial_circuit_compilation.py

```diff
@@ -0,0 +1,242 @@
+r"""Compilation of quantum circuits
+===============================
+
+.. meta::
+    :property="og:description": Learn about circuit transformations and quantum circuit compilation with PennyLane
+    :property="og:image": https://pennylane.ai/qml/_images/thumbnail_tutorial_circuit_compilation.png
+
+.. related::
+
+   tutorial_quantum_circuit_cutting Introduction to circuit cutting
+
+
+*Author: Borja Requena — Posted: 10 May 2023.*
+
+Quantum circuits take many different forms from the moment we design them to the point where they
+are ready to run in a quantum device. These changes happen during the compilation process of the
+quantum circuits, which rely on the fact that there are many equivalent representations of quantum
+circuits that use different sets of gates but provide the same output.
+
+Out of those representations, we are typically interested in finding the most suitable one for our
+purposes. For example, we usually look for the one that will incur the least amount of errors in the 
+specific hardware for which we are compiling the circuit. This usually implies decomposing the
+quantum gates in terms of the native ones of the quantum device, adapting the operations to the
+hardrware's topology, combining them to reduce the circuit depth, etc.
+
+A great part of the compilation process consists on repeatedly performing minor circuit modifications.
+In PennyLane, we can apply :mod:`~pennylane.transforms` to our quantum functions in order to obtain
+equivalent ones that may be more convenient for our task. In this tutorial, we introduce the most
+fundamental transforms involved in the compilation of quantum circuits.
+
+Circuit transforms
+------------------
+
+When we implement quantum algorithms, it is typically in our best interest that the resulting
+circuits are as shallow as possible, especially with the noisy quantum devices available at present.
```

Comment:
```suggestion
circuits are as shallow as possible, especially with the noisy quantum devices available presently.
```

Or "currently", or even "at the time of writing" if you wanna be extra fancy 🎩.